packages/runtime/container-runtime/src/dataStoreContext.ts

```diff
@@ -620,24 +637,44 @@ export class RemotedFluidDataStoreContext extends FluidDataStoreContext {
                     throw new Error(`Invalid snapshot format version ${snapshotFormatVersion}`);
                 }
                 this.pkg = pkgFromSnapshot;
-                isRootStore = isRootDataStore;
             }
 
-            /**
-             * If there is no isRootDataStore in the attributes blob, set it to true. This will ensure that
-             * data stores in older documents are not garbage collected incorrectly. This may lead to additional
-             * roots in the document but they won't break.
-             */
             this.details = {
                 // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
                 pkg: this.pkg!,
-                isRootDataStore: isRootStore ?? true,
                 snapshot: tree ?? undefined,
             };
         }
 
         return this.details;
     }
+
+    protected async getInitialGCDetails(): Promise<IGCDetails> {
+        if (this.initialGcDetails === undefined) {
+            /**
+             * Default value of initial GC details in case the initial snapshot does not have GC details blob.
+             * isRootNode is set to true so that data stores in older documents are not collected by GC. This
+             * may lead to additional root nodes but they won't break.
+             */
+            let gcDetails: IGCDetails = {
+                isRootNode: true,
+                gcNodes: [],
+            };
+
+            /**
+             * If the initialSnapshotValue is null or of type string, the snapshot is in old format. So, it won't have
+             * GC details.
+             */
+            if (!(this.initSnapshotValue === null || typeof this.initSnapshotValue === "string")) {
+                const tree: ISnapshotTree = await this.initSnapshotValue;
```

Comment:
As per my discussion with @arinwt, in case of differential summary, we need to decode the summary by calling decodeSummary in SummarizerNodeUtils. I will do that separately since it will need additional changes to make sure we do it in a consistent way.

Reply:
Note that differential summaries are currently fully disabled by PR #4285, because of issue #4296.

Reply:
I'm also not convinced they should even come back fully, as they too easily hide issues. If we do keep them, i think we'll need to be much stricter with them, like forcing a document loaded via differential summary to be read-only until it sucessfully summarzies.

Reply:
Hmmm. I will hold on the change to handle this scenario here then. Should we create a new issue to track this? Or add a note to #4296 to handle this scenario too? @arinwt @anthony-murphy

Reply:
Either way is fine but definitely add a note to that issue or add a new issue.